### PR TITLE
Chore/prettier

### PR DESCRIPTION
**Pull request by bolmaster2**
- Adds prettier with config `@dreamstechnology/prettier-config-dreams`
- Run prettier with lefthook on pre-commit and in ci workflow
- Add `.node-version` file with v16 (bumped from v14)

### TODO

- [x] Remove the authentication to `@dreamstechnology/prettier-config-dreams` when it's available as public npm package

### CODE REVIEW

File: .envrc.example [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Required by .npmrc
+# Use your own personal GitHub Token with `read:packages` and `repo` access rights.
+# GitHub has excellent guides for how to create such a token, see
+# https://help.github.com/en/articles/creating-a-personal-access-token-for-the-command-line
```
[the remark this answers is not in the document]
Comment (bolmaster2) [reply .envrc.example:4]: I actually like having this info about this as close as possible to where it's used. This is copy-pasted from here: https://github.com/dreamstechnology/dreams-enterprise/blob/main/.envrc.example#L5-L9. I'll add it to the README as well!

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -9,6 +9,28 @@ on:
...
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          persist-credentials: false
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14.x'
```
[the remark this answers is not in the document]
Comment (bolmaster2) [reply .github/workflows/ci.yml:25]: Good point! Let's update it

File: .node-version [hunk 3]
```diff
@@ -0,0 +1 @@
+14.19.2
```
[the remark this answers is not in the document]
Comment (bolmaster2) [reply .node-version:1]: It's used locally by both nvm and asdf depending what you use... I think the node-version is a better name than nvm as it's not tied to a specific tool as nvm - it just says which node version we use. I can add to output this version in the CI as we do in DES

File: .github/workflows/ci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (bolmaster2) [reply .github/workflows/ci.yml]: The tests failed when using v18, but worked with v16. Let's use that as it's the active LTS?

**Comment by bolmaster2** [comment]
I converted this to a draft and added a TODO in the description while we wait for the `@dreamstechnology/prettier-config-dreams` to be publicly accessible. I forgot that 🙈

File: .github/workflows/ci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (thelitek) [reply .github/workflows/ci.yml]: Yep, it's still better than 14.

File: .envrc.example (hunk 1 above)
[the remark this answers is not in the document]
Comment (thelitek) [reply .envrc.example:4]: The jury is still out wether we will need it for the public package: if we go with github for hosting it a token is still needed (no, I don't know what is the point of it).

File: .node-version (hunk 3 above)
[the remark this answers is not in the document]
Comment (thelitek) [reply .node-version]: I thought you intended to use it in a workflow. The mere presence of the file is a value added for whoever want's to play with the repo.

**Review by thelitek — APPROVED** [review]

**Review by thelitek — APPROVED** [review]

**Pull request merged**